src/filters.rs

```diff
@@ -60,6 +60,46 @@ pub fn upcase(input: &Value, _args: &[Value]) -> FilterResult {
     }
 }
 
+
+pub fn downcase(input: &Value, _args: &[Value]) -> FilterResult {
+    match *input {
+        Str(ref s) => Ok(Str(s.to_lowercase())),
+        _ => Err(InvalidType("String expected".to_owned())),
+    }
+}
+
+
+pub fn capitalize(input: &Value, _args: &[Value]) -> FilterResult {
+    match *input {
+        Str(ref s) => Ok(Str(s.char_indices().fold(String::new(), |word, (_, chr)| {
+            let next_char = match word.chars().last() {
+                Some(last) =>
+                    if last.is_whitespace() {
+                        chr.to_uppercase().next().unwrap()
+                    } else {
+                        chr
+                    },
+                _ => chr.to_uppercase().next().unwrap(),
+            }.to_string();
+            word + &next_char
```

Comment:
> Re Boolean for tracking: indeed, that could be done. That's how I used to do that in python, but never liked it. In case in particular as my otherwise very functional closure has an ugly side effect it constantly depends on.

That doesn't require any side effects, the accumulator variable of the fold can just be `(String, bool)` instead of `String`. Although I have to say that even as a Haskell programmer, I'm not sure that a fold is really better than a loop here.